md/src/main/java/io/codelex/joiner/Joiner.java

```diff
@@ -0,0 +1,23 @@
+package io.codelex.joiner;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+public class Joiner <T> {
+
+    private List<T> items;
+    private BiFunction <String, T, String> function;
+
+
+    public Joiner(String separator, BiFunction<String, T, String> function) {
```

Comment:
Šis ir uztaisīts sarežģītāks nekā bija nepieciešams - konsturktorā funkcija nebija jāpadod, tāpat arī attiecīgi klases mainīgais function ir lieks.